eldritch/test_combat.py

```diff
@@ -1978,5 +1978,108 @@ def testDeactivatesOnCombatLoss(self):
+    ):
+      fight_or_flee = self.resolve_to_choice(MultipleChoice)
+      fight_or_flee.resolve(self.state, "Fight")
+
+      return self.resolve_to_choice(CombatChoice)
+
+  def testSanePass(self):
+    self.char.sanity = 4
+    self.char.fight_will_slider = 0
+    worm = monsters.GiantWorm()
+    self.start(worm)
+    self.assertEqual(self.char.sanity, 3)
+    self.assertEqual(self.char.stamina, 5)
+    self.state.event_stack.append(self.state.usables[0]["Bind Monster0"])
+    choose_ignore = self.resolve_to_choice(SpendChoice)
+    self.spend("sanity", 2, choose_ignore)
+    choice = self.resolve_to_choice(SpendChoice)
+    choice.resolve(self.state, "Cast")
+    with mock.patch.object(
+        events.random, "randint", new=mock.MagicMock(return_value=5)
+    ):
+      choice = self.resolve_to_choice(CombatChoice)
+
+    choice.resolve(self.state, "done")
+    self.resolve_until_done()
+    self.assertIn(worm, self.char.trophies)
+    self.assertEqual(self.char.sanity, 1)
+    self.assertEqual(self.char.stamina, 4)  # overwhelming
+    self.assertNotIn(self.bind_monster, self.char.possessions)
+
+  def testInsanePass(self):
+    self.char.sanity = 3
+    self.char.fight_will_slider = 0
+    worm = monsters.GiantWorm()
+    self.start(worm)
+    self.assertEqual(self.char.sanity, 2)  # Lost one to nightmarish
+    self.assertEqual(self.char.stamina, 5)
+    self.state.event_stack.append(self.state.usables[0]["Bind Monster0"])
+    choose_ignore = self.resolve_to_choice(SpendChoice)
+    self.spend("sanity", 2, choose_ignore)
+    choice = self.resolve_to_choice(SpendChoice)
+    choice.resolve(self.state, "Cast")
+    with mock.patch.object(
+        events.random, "randint", new=mock.MagicMock(return_value=5)
+    ):
+      insane_choice = self.resolve_to_choice(ItemLossChoice)
+    insane_choice.resolve(self.state, "done")
+
+    self.resolve_until_done()
+    self.assertIn(worm, self.char.trophies)
+    self.assertEqual(self.char.place.name, "Asylum")
+    self.assertNotIn(self.bind_monster, self.char.possessions)
+    # self.assertEqual(self.char.stamina, 4) # Overwhelming 1
+    # TODO: Overwhelming should also happen
+
+  def testSaneFail(self):
+    self.char.sanity = 3
+    self.char.fight_will_slider = 0
+    self.char.possessions.append(items.TommyGun(0))
+    cultist = monsters.Cultist()
+    self.start(cultist)
+    self.assertEqual(self.char.sanity, 3)  # Lost one to nightmarish
+    self.state.event_stack.append(self.state.usables[0]["Bind Monster0"])
+    choose_ignore = self.resolve_to_choice(SpendChoice)
+    self.spend("sanity", 2, choose_ignore)
+    choice = self.resolve_to_choice(SpendChoice)
+    choice.resolve(self.state, "Cast")
+    with mock.patch.object(
+        events.random, "randint", new=mock.MagicMock(return_value=4)
+    ):
+      choice = self.resolve_to_choice(CombatChoice)
+
+    with self.assertRaises(AssertionError):
+      # We already used our hands
+      choice.resolve(self.state, "Tommy Gun0")
+
```

Comment:
There's a few more assertions that we should make here:
(a) after we failed to cast the spell, it should no longer be usable
(b) after one round of combat, where we presumably roll all 1s and fail to defeat the cultist (...hypothetically), we should be able to deactivate the spell and use our hands again.